opensearch_py_ml/ml_commons/model_uploader.py

```diff
@@ -178,7 +181,11 @@ def _check_mandatory_field(self, model_meta: dict) -> bool:
             if not model_meta.get(TOTAL_CHUNKS_FIELD):
                 raise ValueError(f"{TOTAL_CHUNKS_FIELD} can not be empty")
             if not model_meta.get(MODEL_CONFIG_FIELD):
-                raise ValueError(f"{MODEL_CONFIG_FIELD} can not be empty")
+                if not (
+                    model_meta.get(MODEL_FUNCTION_NAME) == "SPARSE_ENCODING"
+                    or model_meta.get(MODEL_TASK_TYPE) == "SPARSE_ENCODING"
+                ):
```

Comment:
The nested if query is not straightforward. I would suggest switch the nested order or merge two of them together.

Reply:
```
if not model_meta.get(MODEL_CONFIG_FIELD):
    if model_meta.get(MODEL_FUNCTION_NAME) != "SPARSE_ENCODING" and model_meta.get(MODEL_TASK_TYPE) != "SPARSE_ENCODING":
        raise ValueError(f"{MODEL_CONFIG_FIELD} cannot be empty")
```

more clearer I think